pkg/controller/nodeipam/ipam/range_allocator.go

```diff
@@ -192,8 +192,10 @@ func (r *rangeAllocator) worker(stopChan <-chan struct{}) {
 				return
 			}
 			if err := r.updateCIDRsAllocation(workItem); err != nil {
-				// Requeue the failed node for update again.
-				r.nodeCIDRUpdateChannel <- workItem
+				if !apierrors.IsNotFound(err) {
```

Comment:
Thinking that `updateCIDRsAllocation` should have a second return value  `shouldRetry bool` as opposed to inspecting the error here. What do you think?

Reply:
On second thought, I think if node is not found in `updateCIDRsAllocation`, we should attempt to release the cidr similar to what is done on line 385 - 392:

https://github.com/kubernetes/kubernetes/blob/82738f15cfe22b2aca2de2f1a78ef87f271684fb/pkg/controller/nodeipam/ipam/range_allocator.go#L385-L392